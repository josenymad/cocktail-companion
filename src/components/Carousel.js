import { React, useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowLeft, faArrowRight } from "@fortawesome/free-solid-svg-icons";
import carouselData from "./CarouselData";
import "../styles/carousel.css";

const Carousel = ({ items }) => {
  const [currentImage, setCurrentImage] = useState(0);
  const length = { items };

  const nextImage = () => {
    setCurrentImage(currentImage === length - 1 ? 0 : currentImage + 1);
  };

  const previousImage = () => {
    setCurrentImage(currentImage === 0 ? length - 1 : currentImage - 1);
  };

  if (!length) {
    return null;
  }

  return (
    <div className="carousel">
      <FontAwesomeIcon
        className="left-arrow"
        icon={faArrowLeft}
        onClick={previousImage}
      />
      <FontAwesomeIcon
        className="right-arrow"
        icon={faArrowRight}
        onClick={nextImage}
      />
      {carouselData.map((item, index) => {
        return (
          <div
            className={index === currentImage ? "image-active" : "image"}
            key={item.alt}
          >
            {index === currentImage && (
              <img src={item.image} alt={item.alt} className="image" />
            )}
          </div>
        );
      })}
    </div>
  );
};

export default Carousel;
