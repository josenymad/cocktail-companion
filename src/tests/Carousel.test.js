import React from "react";
import { render, screen } from "@testing-library/react";
import Carousel from "../components/Carousel";

describe("Carousel", () => {
  const validProps = {
    selectedDrink: ["mock-property"],
    searchQuery: "mock-drink",
    setSelectedDrink: jest.fn(),
    drinks: [
      {
        idDrink: "1",
        strDrinkThumb: "mock-url1",
        strDrink: "mock-title1",
      },
      {
        idDrink: "2",
        strDrinkThumb: "mock-url2",
        strDrink: "mock-title2",
      },
    ],
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <Carousel
        selectedDrink={validProps.selectedDrink}
        searchQuery={validProps.searchQuery}
        setSelectedDrink={validProps.setSelectedDrink}
        slides={validProps.drinks}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct image", () => {
    render(
      <Carousel
        selectedDrink={validProps.selectedDrink}
        searchQuery={validProps.searchQuery}
        setSelectedDrink={validProps.setSelectedDrink}
        slides={validProps.drinks}
      />,
    );

    expect(screen.getByAltText("cocktail")).toBeInTheDocument();
  });

  it("renders the correct title", () => {
    render(
      <Carousel
        selectedDrink={validProps.selectedDrink}
        searchQuery={validProps.searchQuery}
        setSelectedDrink={validProps.setSelectedDrink}
        slides={validProps.drinks}
      />,
    );

    expect(screen.getByText("mock-title1")).toBeInTheDocument();
  });
});
