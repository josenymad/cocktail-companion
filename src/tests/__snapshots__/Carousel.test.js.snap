// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders correctly 1`] = `
<DocumentFragment>
  <div
    class="carousel"
  >
    <div
      class="slide active"
    >
      <div
        class="cocktail-card"
      >
        <button
          class="cocktail-card__button"
          type="button"
        >
          <img
            alt="thumbnail"
            class="cocktail-card__thumbnail"
            src="mock-url1"
          />
          <p
            class="cocktail-card__name"
          >
            mock-title1
          </p>
        </button>
      </div>
    </div>
    <div
      class="slide"
    />
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-arrow-left carousel__left-arrow"
      data-icon="arrow-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
        fill="currentColor"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-arrow-right carousel__right-arrow"
      data-icon="arrow-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
        fill="currentColor"
      />
    </svg>
    <div
      class="cocktail-details"
    >
      <button
        class="cocktail-details__button"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-xmark "
          data-icon="circle-xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM175 175c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        class="cocktail-details__container"
      >
        <img
          alt="cocktail"
          class="cocktail-details__image"
        />
        <div
          class="cocktail-details__text"
        >
          <h3
            class="cocktail-details__name"
          />
          <p
            class="cocktail-details__glass"
          />
          <p
            class="cocktail-details__ingredients"
          >
            Ingredients:
          </p>
          <ul
            class="cocktail-details__ingredients-list"
          />
          <p
            class="cocktail-details__instruction"
          >
            Instructions:
          </p>
          <p
            class="cocktail-details__instructions"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
